bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/SolaxLocalAccessHandler.java

```diff
@@ -0,0 +1,171 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solax.internal;
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.solax.internal.connectivity.LocalHttpConnector;
+import org.openhab.binding.solax.internal.connectivity.rawdata.LocalConnectRawDataBean;
+import org.openhab.binding.solax.internal.model.InverterData;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonParseException;
+
+/**
+ * The {@link SolaxLocalAccessHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Konstantin Polihronov - Initial contribution
+ */
+@NonNullByDefault
+public class SolaxLocalAccessHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SolaxLocalAccessHandler.class);
+
+    private static final int INITIAL_SCHEDULE_DELAY_SECONDS = 5;
+
+    private @NonNullByDefault({}) LocalHttpConnector localHttpConnector;
+
+    private @Nullable ScheduledFuture<?> schedule;
+
+    public SolaxLocalAccessHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        SolaxConfiguration config = getConfigAs(SolaxConfiguration.class);
+        localHttpConnector = new LocalHttpConnector(config.password, config.hostname);
+        int refreshInterval = config.refreshInterval;
+        TimeUnit timeUnit = TimeUnit.SECONDS;
+
+        logger.debug("Scheduling regular interval retrieval every {} {}", refreshInterval, timeUnit);
+        schedule = scheduler.scheduleWithFixedDelay(this::retrieveData, INITIAL_SCHEDULE_DELAY_SECONDS, refreshInterval,
+                timeUnit);
+    }
+
+    private void retrieveData() {
+        try {
+            String rawJsonData = localHttpConnector.retrieveData();
+            logger.debug("Raw data retrieved = {}", rawJsonData);
+
+            if (rawJsonData != null && !rawJsonData.isEmpty()) {
+                updateData(rawJsonData);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "JSON data could not be retrieved.");
```

Comment:
I18N could be supported for this text. See https://www.openhab.org/docs/developer/utils/i18n.html#using-custom-keys

Reply:
Hope I got it right. Please verify. Not sure if the bindingName is supposed to be in the .properties file as per the document. I added it as well. If I regenerate the channels again will the custom values get persisted?

Reply:
Sorry, the documentation was slightly out of sync and thus misleading. I've just fixed the part about bindingName: openhab/openhab-docs#2117. Besides that, your implementation looks correct. You can also verify by changing your code so that line would be run. 🙂